src/roiextractors/extractors/tiffimagingextractors/scanimagetiffimagingextractor.py

```diff
@@ -891,6 +891,80 @@ def get_frames_per_slice(file_path: PathType) -> int:
         frames_per_slice = non_varying_frame_metadata.get("SI.hStackManager.framesPerSlice", 1)
         return frames_per_slice
 
+    def get_original_frame_indices(self, plane_index: Optional[int] = None) -> np.ndarray:
+        """
+        Get the original frame indices for each sample.
+
+        Returns the index of the original frame for each sample, mapping processed samples
+        back to their corresponding frames in the raw microscopy data. This accounts for
+        any filtering, subsampling, or exclusions (such as flyback frames) performed by
+        the extractor.
+
+        Parameters
+        ----------
+        plane_index : int, optional
+            Which plane to use for frame index calculation in volumetric data.
+            If None, uses the convention of using the last plane as its the default for
+            volumetric data. It must be less than the total number of planes.
+
+        Returns
+        -------
+        np.ndarray
+            Array of original frame indices (dtype: int64) with length equal to the
+            number of samples. Each element represents the index of the original
+            microscopy frame that corresponds to that sample.
+
+        Notes
+        -----
+        **Frame Index Calculation:**
+
+        - **Planar data**: Frame indices are sequential (0, 1, 2, ...)
+        - **Multi-channel data**: Accounts for channel interleaving
+        - **Volumetric data**: Uses the specified plane (default: last plane)
+        - **Multi-file data**: Includes file offsets for global indexing
+        - **Flyback frames**: Automatically excluded from indexing
+
+        **Common Use Cases:**
+
+        - Synchronizing with external timing systems
+        - Mapping back to original acquisition timestamps
+        - Data provenance and traceability
+        - Cross-referencing with raw data files
+
+        **Examples:**
+
+        For a 3-sample volumetric dataset with 5 planes per volume:
+        - Default behavior returns indices [4, 9, 14] (last plane of each volume)
+        - With plane_index=0 returns indices [0, 5, 10] (first plane of each volume)
```

Comment:
This example is a little confusing to me. With plane_index=0, doesn't it return each frame (index) in the first plane?